### PR TITLE
Non-resumable

### LINKED ISSUE #261: Resumable doesn’t belong in the core library
`Resumable` isn’t fully baked—there’s no good story around its control flow, how to support local catching, or whether it should instead be represented with coroutines[1] or `Lift`[2] or general effects[3].

For 1.0, we really only want to have things in the core library that are bulletproof, so let’s move it to its own library instead. It can come back if and when we’ve got a better story for it.

1: Coroutines provide a similarly general request-and-response model; through that lens, `Resumable` can be viewed as a type-indexed coroutine (i.e. requests can specify the type to resume with). cf https://hackage.haskell.org/package/freer-simple-1.2.1.1/docs/Control-Monad-Freer-Coroutine.html
2: The error parameter to `Resumable` is of kind `* -> *`, just like the parameter to `Lift`; the only difference being that `Lift` holds `err (m k)` whereas `Resumable` splits this into `(err a)` and `(a -> m k)` for some existentially-quantified `a`. I believe anything representable with the latter is representable with the former, possibly with some slight restructuring to accommodate a `Functor` instance.
3: The `Resumable.Resume` carrier works very much like `InterpretC` used to, simply using a higher-order function to interpret the request. This can be handled perfectly well with `InterpretC` on its own, with the added bonus that the exceptions can embed actions if they so desire. (If this extra flexibility is undesirable, see [2].) Behaviour like `Resumable.Either` can further be provided by a carrier suspending requests to _any_ effect, while additionally accommodating resumption after the fact.

### PR DESCRIPTION
**Pull request by robrix**
This PR 🔥s `Resumable`.

- [x] Fixes #261.

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -52,6 +52,8 @@
 
 - Redefines `Fail` as a synonym for `Throw String`. ([#247](https://github.com/fused-effects/fused-effects/pull/247))
 
+- Removes `Resumable` and its carriers, as they can be more usefully and flexibly replaced by arbitrary effects, `Lift`, and `InterpretC`. ([#269](https://github.com/fused-effects/fused-effects/pull/269))
```
Comment (robrix) [inline ChangeLog.md]: I’m honestly debating whether we need to re-home it or whether just removing it is fine.

We’ll probably want to copy it into `semantic` as a short-term migration thing, but I don’t think it really _deserves_ its own package—tho better options (e.g. coroutines) might.

Comment (patrickt) [reply ChangeLog.md]: I think we should move it to a package, and when we land a better solution (e.g. coroutines) we can mark the package as deprecated. That’s the most graceful solution, in my view, given that I bragged about `Resumable` at Strange Loop and I don’t want to renege on that particular promise. What do you think?

Comment (robrix) [reply ChangeLog.md]: Yeah, that’s a good call 👍

Comment (robrix) [reply ChangeLog.md]: Yeah, that’s a good call 👍

Comment (robrix) [reply ChangeLog.md]: Gonna hold off merging this PR until we have a repo & URL for that.

**Pull request merged**